python/lsst/ts/atwhitelight/atwhitelight_csc.py

```diff
@@ -0,0 +1,446 @@
+                f"> high_supply_temperature_warning={config.chiller.high_supply_temperature_warning}"
+            )
+        self.config = config
+
+    async def handle_summary_state(self):
+        if self.disabled_or_enabled:
+            if not self.chiller_connected:
+                await self.connect_chiller()
+            if self.summary_state == salobj.State.ENABLED:
+                chiller_watchdog = self.chiller_model.get_watchdog()
+                if chiller_watchdog.alarmsPresent:
+                    await self.fault(
+                        code=ErrorCode.CHILLER_ERROR,
+                        report="Chiller is reporting alarms",
+                    )
+                    return
+            if not self.lamp_connected:
+                await self.connect_lamp()
+            self.should_be_connected = True
+        elif self.summary_state == salobj.State.FAULT:
+            # Turn off the lamp, if connected
+            if self.lamp_connected and self.lamp_model.lamp_on:
+                self.log.warning(
+                    "Going to fault while connected to the lamp controller; "
+                    "forcing the lamp off"
+                )
+                await self.lamp_model.turn_lamp_off(force=True)
+            else:
+                self.log.warning(
+                    "Going to fault and not connect to the lamp controller; "
+                    "please turn the lamp off manually."
+                )
+        else:
+            self.should_be_connected = False
+            if self.lamp_connected and self.lamp_model.lamp_on:
+                await self.lamp_model.turn_lamp_off(force=True)
+            await self.disconnect_lamp()
+            await self.disconnect_chiller()
+
+    async def connect_chiller(self):
+        await self.disconnect_chiller()
+        if self.chiller_model is None:
+            self.chiller_model = ChillerModel(
+                config=self.config.chiller,
+                topics=self,
+                log=self.log,
+                status_callback=self.status_callback,
+                simulate=self.simulation_mode != 0,
+            )
+        await self.chiller_model.connect()
+
+    async def connect_lamp(self):
+        await self.disconnect_lamp()
+        if self.lamp_model is None:
+            labjack_config = self.config.lamp
+            self.lamp_model = LampModel(
+                config=labjack_config,
+                topics=self,
+                log=self.log,
+                status_callback=self.status_callback,
+                simulate=self.simulation_mode != 0,
+            )
+        await self.lamp_model.connect()
+
+    async def disconnect_chiller(self):
+        # Don't use chiller_connected because that can be false
+        # even if a basic connection exists (before status seen)
+        if self.chiller_model is None:
+            return
+        await self.chiller_model.disconnect()
```

Comment:
Perhaps you should set chiller_model to None after disconnect? I don't see any other "= None" except in the constructor, this will prevent the disconnect/connect sequence to work

Reply:
The lamp model has two important timers that I don't want to lose when disconnected. So I'd rather keep both models around (for symmetry; I could probably null the chiller model without ill effect).